docs/triggers/trigger-generic-inputs.md

```diff
@@ -0,0 +1,80 @@
+# Trigger Creation Workflow Definition Inputs
+
+When creating a trigger at runtime, a workflow definition can be passed into the `client.workflows.triggers.create` method which will allow for typeahead/autocomplete of the input parameters based on the definition's inputs. 
+
+## Defining A Workflow Recap
+
+Recall in the `manifest.ts` file of an app, a `workflow` can be defined using the `DefineWorkflow` method as follows:
+
+```ts
+export const TriggersWorkflow = DefineWorkflow({
+  callback_id: "triggers_workflow",
+  title: "Triggers Workflow",
+  description: "sample runthrough of a trigger creation",
+  input_parameters: {
+    properties: {
+      a_string: {
+        type: Schema.types.string,
+      },
+      b_string: {
+        type: Schema.types.string,
+      },
+      a_channel: {
+        type: Schema.slack.types.channel_id,
+      },
+    },
+    required: [],
+  },
+});
+```
+
+When a workflow is created in this manner, if contains a `workflow.definition` object. By passing this definition into the `create` method, the trigger will have access to information related to the workflow's definition.
+
+### Using Workflow definition input at runtime Example 
+
+```ts
+import type { SlackFunctionHandler } from "../../deno-slack-sdk/src/types.ts";
+import { TriggersFunction, TriggersWorkflow } from "../manifest.ts";
+import { SlackAPI } from "../../deno-slack-api/src/mod.ts";
+
+const triggers: SlackFunctionHandler<typeof TriggersFunction.definition> = 
+  async (
+    { token, env },
+  ) => {
+    const client = SlackAPI(token, {
+      slackApiUrl: env["SLACK_API_URL"],
+    });
+    const workflowDef = TriggersWorkflow.definition;
+    const triggerResponse = await client.workflows.triggers.create<
+      typeof workflowDef //Definition passed in as a generic
+    >({
```

Comment:
```suggestion
    const triggerResponse = await client.workflows.triggers.create<
      typeof workflowDef.definition //Definition passed in as a generic
    >({
```

Is there a reason we'd recommend them to instantiate a variable for the workflow definition rather than doing it directly in the generic? My two cents is that accessing the definition directly in the generic should be the recommended way